docs/maintainer/understanding_conda_forge/feedstocks.md

```diff
@@ -0,0 +1,184 @@
+# The feedstock
+
+A feedstock is the home of one recipe; it may produce more than one conda artifact in the case of a multi-output recipe.
+The feedstock is the stage for most events about a recipe.
+
+## Initialization
+
+The feedstock repository is created following the merging in `staged-recipes` (see [Feedstock creation](staged_recipes.md#feedstock-creation) for details on that process).
+
+## Package building diagram
+
+Package building can be triggered by several events, which are described in the next section.
+In all of those cases, the following sequence plays out.
+
+```mermaid
+sequenceDiagram
+    box github
+    participant f as feedstock
+    end
+    box CI provider
+    participant c as CI
+    end
+    box distribution infrastructure
+    participant s as cf-staging channel
+    participant cf as conda-forge channel
+    participant cdn as CDN
+    end
+    f->>+c: trigger build
+    Note right of c: 1. build<br/>2. validate
+    opt if valid and required (4.)
+    c->>s: upload packages
+    s->>c: report reception success/failure
+    Note right of s: 3. validate
+    s->>cf: upload packages
+    cf->>cdn: upload packages
+    end
+    c->>-f: report success/failure
+```
+
+:::note
+
+1. For Linux and `noarch` packages, the build itself is carried out in a Docker container on the CI. On macOS and Windows, the CI runner system image is used after tuning it a bit.
+2. The validation checks that the artifacts produced during the build are permissible for this feedstock by consulting [the `feedstock-outputs` repository](/docs/maintainer/infrastructure/#feedstock-outputs).
+3. The server-side validation is essentially the same as 2. It is repeated to guard against potential intentional or unintentional interference at the feedstock level, which is easier to access.
+4. The upload is only triggered if the validation (3.) was successful. Additionally, it is only performed under certain conditions, for example for commits in `main`, but _not_ in PRs.
+
+:::
+
+## Events that trigger new builds
+
+Package building is either triggered as part of a PR or on commits to branches in the feedstock repository (which in turn usually stem from the merging of PRs).
+
+:::warning
+_Any_ commit on _any_ branch in the feedstock can lead to the building and publication of a package.
+To avoid the gratuitous publication of improper packages, development branches MUST NOT be added to the feedstock repository.
+Instead, they live in forks of the feedstock repository and the associated work is added only via pull requests.
+:::
```

Comment:
Thank you!  I've been meaning to add something like this, so we can link in PRs, forever now but never got it.